main.py

```diff
@@ -0,0 +1,71 @@
+import os
+
+from mako.template import Template
+from mako.lookup import TemplateLookup
+
+from question_groups import admin, exec, finissante
+from models.survey import Survey
+
+from config import config
+
+for group in config.groups:
+    if not os.path.isfile(f"input/{group.input}"):
+        print(f"Check if {group.input} is present in input directory")
+        exit()  # TODO error code?
+
+if config.type == "l'AGEG" and not all([g.type == "admin" or g.type == "exec" for g in config.groups]):
+    print(f"Elections for l'AGEG must only contain exec or admin question groups, see GroupConfig in config.py")
+    exit()
+
+if config.type == "la finissante" and (len(config.groups) > 1 or not all([g.type == "finissante" for g in config.groups])):
+    print(f"Elections for la finissante must only contain one finissante question group, see GroupConfig in config.py")
+    exit()
+
+print(f"JOB START ! Starting generation for {config.type}!")
+groups = []
+questions = []
+sousquestions = []
+
+for i, group in enumerate(config.groups):
+    if group.type == "exec":
+        group, question = exec.generate_questions(group)
+        groups.append(group)
+        questions.extend(question)
+
+    elif group.type == "admin":
+        group, question, sousquestion = admin.generate_questions(group)
+        groups.append(group)
+        questions.append(question)
+        sousquestions.extend(sousquestion)
+    elif group.type == "finissante":
+        group, question = finissante.generate_questions(group)
+        groups.extend(group)
+        questions.extend(question)
+    else:
+        print("invalid conf type, exiting")
+        exit()  # TODO discard conf?
+
+survey_options = Survey(config)
+
+mylookup = TemplateLookup(directories=['.'], input_encoding="utf-8", output_encoding="utf-8")
+mytemplate = Template(filename='templates/base.mako', lookup=mylookup, input_encoding="utf-8", output_encoding="utf-8")
+
+if config.type == "l'AGEG":
```

Comment:
J'aurais utilisé des constances pour le Type il est réutilisé souvant.

Reply:
C'est noté, je rajouterai ces constantes dans une itération future. Probablement avec des enum.